lib/internal/modules/esm/get_format.js

```diff
@@ -66,6 +72,16 @@ function extname(url) {
   return '';
 }
 
+/**
+ * Determine whether the given file URL is under a `node_modules` folder.
+ * This function assumes that the input has already been verified to be a `file:` URL,
+ * and is a file rather than a folder.
+ * @param {URL} url
+ */
+function underNodeModules(url) {
+  return StringPrototypeIndexOf(url.pathname, '/node_modules/') !== -1;
```

Comment:
Should we check the scheme of the URL? I assume we would do that check only for file: URL, right?

Reply:
That’s a good point. I think we at least want to exclude `data:` URLs:

```shell
> u = new URL('data:text/javascript,console.log(3)')
URL {
  href: 'data:text/javascript,console.log(3)',
  origin: 'null',
  protocol: 'data:',
  username: '',
  password: '',
  host: '',
  hostname: '',
  port: '',
  pathname: 'text/javascript,console.log(3)',
  search: '',
  searchParams: URLSearchParams {},
  hash: ''
}
```

I’m less confident about `http:` URLs. Presumably if an HTTPS URL includes `node_modules`, it’s just a served version of a traditional Node “on disk” app and the same resolution rules should apply. How does `--experimental-network-imports` work today? If a module loaded via HTTPS includes `import('pkg')`, does Node look for `pkg` under a `/node_modules` path under that HTTPS URL, or does Node look in the filesystem, or neither? However `--experimental-network-imports` currently implements the ESM resolution algorithm should probably dictate what we do here. cc @bmeck

Reply:
I’ll limit this to files since I’m pretty sure that for both `data:` and `https:` URLs we determine type based on MIME type headers rather than anything on disk (like `package.json` files).